pmlc/dialect/tile/builder.cc

```diff
@@ -693,6 +694,7 @@ std::shared_ptr<TileProgram> TileBuilder::MakeProgram(StringRef name, const Prog
   // Do some optimization passes
   mlir::PassManager pm(&impl->context);
   pm.addPass(MakeProgramPass::create());
+  pm.addPass(dialect::tile::createConstantTypesPass(mutations.floatx, mutations.intx));
```

Comment:
Since this translation unit is in the `pmlc::dialect::tile` namespace, there's no need to prefix with `dialect::tile::`

Reply:
This pass needs to run before the initial `MakeProgramPass` because we want constant folding to use the updated constant datatype.